src/data/sum.lean

```diff
@@ -93,6 +93,24 @@ protected def elim {α β γ : Sort*} (f : α → γ) (g : β → γ) : α ⊕ 
 @[simp] lemma elim_inr {α β γ : Sort*} (f : α → γ) (g : β → γ) (x : β) :
   sum.elim f g (inr x) = g x := rfl
 
+@[simp] lemma elim_comp_inl {α β γ : Sort*} (f : α → γ) (g : β → γ) :
+  sum.elim f g ∘ inl = f := rfl
+  
+@[simp] lemma elim_comp_inr {α β γ : Sort*} (f : α → γ) (g : β → γ) :
+  sum.elim f g ∘ inr = g := rfl
+  
+@[simp] lemma elim_inl_inr {α β : Sort*} :
+  @sum.elim α β _ inl inr = id :=
+funext $ λ x, sum.cases_on x (λ _, rfl) (λ _, rfl)
+
+lemma comp_elim {α β γ δ : Sort*} (f : γ → δ) (g : α → γ) (h : β → γ):
```

Comment:
This would also be fine as a simp lemma, but I'm not sure if we actually want this to always simplify to the right-hand side.

Reply:
Yeah, I left this one out because the decision of which way to simplify felt subjective.